src/v/rpc/transport.cc

```diff
@@ -192,33 +193,41 @@ transport::do_send(sequence_t seq, netbuf b, rpc::client_opts opts) {
       });
 }
 
-void transport::dispatch_send() {
+void transport::dispatch_send(const client_opts& opts) {
     ssx::background
-      = ssx::spawn_with_gate_then(_dispatch_gate, [this]() mutable {
-            return ss::do_until(
-              [this] {
-                  return _requests_queue.empty()
-                         || _requests_queue.begin()->first
-                              > (_last_seq + sequence_t(1));
-              },
-              [this] {
-                  auto it = _requests_queue.begin();
-                  _last_seq = it->first;
-                  auto buffer = std::move(it->second->buffer).get();
-                  auto units = std::move(it->second->resource_units);
-                  auto v = std::move(*buffer).as_scattered();
-                  auto msg_size = v.size();
-                  _requests_queue.erase(it->first);
-                  return _out.write(std::move(v))
-                    .finally([this, msg_size, units = std::move(units)] {
-                        _probe.add_bytes_sent(msg_size);
-                    });
-              });
-        }).handle_exception([this](std::exception_ptr e) {
-            vlog(rpclog.info, "Error dispatching socket write:{}", e);
-            _probe.request_error();
-            fail_outstanding_futures();
-        });
+      = ssx::spawn_with_gate_then(
```

Comment:
can you split this commit into two where: 
1) would apply change to the dispatch logic
2) would change formatting

Reply:
oh, sorry i think i wasn't clear, i meant this `dispatch_send` method, it doesn't really change but the indentation does making it hard to track actual changes.

Reply:
Oh no worries, now that I removed the timeout handling, I don't think indents is an issue anymore? Also I guess I could just retain the coroutinization change, seems like the direction we generally want to move in for code hygeine?

Reply:
@mmaslankaprv Now that we have a release branch, I updated the PR with your suggestion..The ordering test that we chatted about is exactly the same one in rpc_gen_cycling_test::`ordering_test`, so didn't add a new one.